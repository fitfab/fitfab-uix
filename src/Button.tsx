export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {}
export const Button = ({ children, ...props }: ButtonProps) => {
  return <button {...props}>{children}</button>;
};
