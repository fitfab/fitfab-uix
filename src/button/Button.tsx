export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  primary?: boolean;
}
export const Button = ({ children, primary, ...props }: ButtonProps) => {
  const variant = primary
    ? "bg-black hover:bg-neutral-800"
    : "bg-slate-400 hover:bg-slate-600";
  const btnClasses = `text-clip text-white uppercase px-4 flex justify-center items-center min-w-[5rem] max-w-xs h-10 rounded-full ${variant}`;
  return (
    <button className={btnClasses} {...props}>
      {children}
    </button>
  );
};
