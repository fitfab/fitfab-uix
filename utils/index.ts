export const hasClass = (el: HTMLElement | any, cssClass: string): boolean => {
  return el.getAttribute('class').split(' ').includes(cssClass)
}
