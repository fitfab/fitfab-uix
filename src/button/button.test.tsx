import { render, screen } from '@testing-library/react'
import { hasClass } from '../../utils'
import { Button } from './Button'

it('renders correctly', () => {
  const { container } = render(
    <Button>Facebook</Button>
  )
  expect(container).toMatchSnapshot()
  const copy = screen.getByText('Facebook')
  expect(copy).toBeVisible()
  expect(copy).toHaveTextContent(/Facebook/i)
})

it('renders primary button correctly', () => {
  render(
    <Button primary>read</Button>
  )
  const button = screen.getByText('read')
  expect(hasClass(button, 'bg-black')).toBe(true)
})

it('renders secondary button correctly', () => {
  render(
    <Button primary={false}>learn</Button>
  )
  const button = screen.getByText('learn')
  expect(hasClass(button, 'bg-slate-500')).toBe(true)
})
