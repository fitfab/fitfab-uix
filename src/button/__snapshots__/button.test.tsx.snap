// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    class="text-white tracking-wide uppercase px-4 truncate min-w-[5rem] max-w-xs h-10 rounded-full bg-black hover:bg-neutral-800"
  >
    Facebook
  </button>
</div>
`;
